src/components/SkillCreator/SkillCreator.js

```diff
@@ -33,95 +33,383 @@ import ChevronRight from 'material-ui/svg-icons/navigation/chevron-right';
 // Ant Design Components
 import notification from 'antd/lib/notification';
 import Icon from 'antd/lib/icon';
+import { red500 } from 'material-ui/styles/colors';
 
 const cookies = new Cookies();
 const languages = [];
+const fontsizes = [];
+const codeEditorThemes = [];
 let self;
 
 export default class SkillCreator extends Component {
   constructor(props) {
     super(props);
+
     let skillBuildCode =
       '::name <Skill_name>\n::category <Category>\n::language <Language>\n::author <author_name>\n::author_url <author_url>\n::description <description> \n::dynamic_content <Yes/No>\n::developer_privacy_policy <link>\n::image <image_url>\n::terms_of_use <link>\n\n\nUser query1|query2|quer3....\n!example:<The question that should be shown in public skill displays>\n!expect:<The answer expected for the above example>\nAnswer for the user query';
     if (this.props.botBuilder) {
       skillBuildCode = this.props.botBuilder.code;
     }
-    this.state = {
-      code: skillBuildCode,
-      codeView: true,
-      conversationView: false,
-      treeView: false,
-      groupSelect: true,
-      languageSelect: true,
-      expertSelect: true,
-      showImage: false,
-      loading: false,
+
+    let commonState = {
       file: null,
-      loadViews: false,
-      imageUrl: '<image_url>',
-      commitMessage: '',
-      modelValue: null,
-      groupValue: null,
-      languageValue: null,
-      expertValue: '',
-      groups: [],
-      slideState: 1, // 1 means in middle, 2 means preview collapsed
-      colSkill: 8,
-      colPreview: 4,
-      prevButton: 0, // 0 means disappear, 1 means appear
       anchorOrigin: {
         horizontal: 'left',
         vertical: 'bottom',
       },
+      groups: [],
+      loadViews: false,
+      code: skillBuildCode,
+      editable: !!cookies.get('loggedIn'),
     };
+
+    if (this.props.location.pathname.split('/')[1] === 'skillCreator') {
+      this.state = {
+        ...commonState,
+        mode: 'create',
+        codeView: true,
+        conversationView: false,
+        treeView: false,
+        showImage: false,
+        loading: false,
+        imageUrl: '<image_url>',
+        commitMessage: '',
+        modelValue: null,
+        groupValue: null,
+        languageValue: null,
+        groupSelect: true,
+        languageSelect: true,
+        expertSelect: true,
+        expertValue: '',
+        slideState: 1, // 1 means in middle, 2 means preview collapsed
+        colSkill: 8,
+        colPreview: 4,
+        prevButton: 0, // 0 means disappear, 1 means appear
+      };
+    } else {
+      if (this.props.hasOwnProperty('revertingCommit')) {
+        this.setState({
+          commitMessage: 'Reverting to commit - ' + this.props.revertingCommit,
+        });
+      } else if (this.props.location.pathname.split('/')[5]) {
```

Comment:
Which case is this? Can you please write inline comments for it

Reply:
This is the case when user is viewing older version of the skill, it was already present in the skillEditor but wasn't implemented properly. I just copied the code before but now i have modified it to work properly. This can be removed if not required.

![screenshot from 2018-11-04 11-37-45](https://user-images.githubusercontent.com/21025052/47960838-11b1a000-e027-11e8-8d6d-7432d408e957.png)